### ISSUE TITLE
validation of params for NewDatabase

### ISSUE THREAD

**Issue by tngTUDOR** [polca/premise#16]
**O**nce upon a time, a young and proud number was sent to the `NewDatabase` function.
The number was `3`.

It all went well, until it didn't: activity data was gotten, exchange data was filled, global scope was set to datasets with missing location, scope of dataset was set to production exchanges with missing location, technosphere exchanges missing locations were corrected, empty exchanges were removed. *But*, oh misery, after starting to add Carma CCS inventories, with a worksheet extracted in less than a second, _Traceback_ the mysterious appeared from the dark. It was his call. _Traceback_ started to throw lines and lines of code, until she got tired, after throwing an `IndexError`.

The leaders from the dimension where `3` was coming, kept sending him, again and again, without success: the `IndexError` kept winning.

...

Centuries went through and finally, the leaders of another dimension sent a young brave number to replace the original warrior. The number was `3.6`.
The sole presence of `3.6` was enough to prevent _Traceback_ the mysterious from appearing, and finally,  Carma CCS inventories were added, along with the fellow fossil carbon dioxide storage technologies, Biogas inventories, Electrolysis Hydrogen inventories, and even the last of the legion, methanol-based synthetic fuel inventories.

### TL;DR:
I suggest `NewDatabase` also validates the actual source_version:
+ `isinstance(source_versio, float)`
+ `source_version in [3.5, 3.6, 3.7]`

```python
>>> _ = NewDatabase(scenario = "SSP2-Base",
... year = 2028,
... source_db = "ecoinvent36-cutoff",
... source_version = 3)
Getting activity data
100%|██████████████████████████████████████| 18121/18121 [00:00<00:00, 95052.80it/s]
Adding exchange data to activities
100%|████████████████████████████████████| 615644/615644 [00:42<00:00, 14529.39it/s]
Filling out exchange data
100%|███████████████████████████████████████| 18121/18121 [00:03<00:00, 4717.18it/s]
Set missing location of datasets to global scope.
Set missing location of production exchanges to scope of dataset.
Correct missing location of technosphere exchanges.
Remove empty exchanges.
Add Carma CCS inventories
Extracted 1 worksheets in 0.48 seconds
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/conda/lib/python3.8/site-packages/rmnd_lca/ecoinvent_modification.py", line 106, in __init__
    self.import_inventories(add_vehicles)
  File "/opt/conda/lib/python3.8/site-packages/rmnd_lca/ecoinvent_modification.py", line 122, in import_inventories
    carma.merge_inventory()
  File "/opt/conda/lib/python3.8/site-packages/rmnd_lca/inventory_imports.py", line 457, in merge_inventory
    self.prepare_inventory()
  File "/opt/conda/lib/python3.8/site-packages/rmnd_lca/inventory_imports.py", line 751, in prepare_inventory
    self.add_product_field_to_exchanges()
  File "/opt/conda/lib/python3.8/site-packages/rmnd_lca/inventory_imports.py", line 545, in add_product_field_to_exchanges
    y["product"] = self.correct_product_field(y)
  File "/opt/conda/lib/python3.8/site-packages/rmnd_lca/inventory_imports.py", line 587, in correct_product_field
    raise IndexError(
IndexError: An inventory exchange in Carma CCS cannot be linked to the biosphere or the ecoinvent database: {'name': 'market for water, completely softened, from decarbonised water, at user', 'amount': 0.006, 'location': 'GLO', 'unit': 'kilogram', 'categories': 'Materials/fuels', 'type': 'technosphere', 'uncertainty type': 2.0, 'loc': -5.115995809754082, 'scale': 0.1682361183106064, 'comment': '(4,5,3,2,3,3); estimate based on literature', 'negative': 0.0, 'simapro name': 'Water, completely softened, at plant/RER U'}
```

**Comment by romainsacchi** [comment]
Should be fixed with your PR.
